src/market/cancel.ts

```diff
@@ -13,19 +14,78 @@ export abstract class OrderCanceler {
      * @returns A promise that resolves when the transaction is confirmed.
      */
     static async cancelOrders(
         providerOrSigner: ethers.providers.JsonRpcProvider | ethers.Signer,
         orderbookAddress: string,
-        orderIds: BigNumber[]
+        orderIds: BigNumber[],
+        txOptions?: TransactionOptions
     ): Promise<ContractReceipt> {
+        console.time('Total Cancel Orders Time');
         try {
             const orderbook = new ethers.Contract(orderbookAddress, orderbookAbi.abi, providerOrSigner);
+            
+            console.time('Get Signer Time');
+            const signer = providerOrSigner instanceof ethers.Signer
+                ? providerOrSigner
+                : providerOrSigner.getSigner();
+            const address = await signer.getAddress();
+            console.timeEnd('Get Signer Time');
+
+            const data = orderbook.interface.encodeFunctionData("batchCancelOrders", [orderIds]);
+
+            const tx: ethers.providers.TransactionRequest = {
+                to: orderbook.address,
+                from: address,
+                data,
+                ...(txOptions?.nonce !== undefined && { nonce: txOptions.nonce }),
+                ...(txOptions?.gasLimit && { gasLimit: txOptions.gasLimit }),
+                ...(txOptions?.gasPrice && { gasPrice: txOptions.gasPrice }),
+                ...(txOptions?.maxFeePerGas && { maxFeePerGas: txOptions.maxFeePerGas }),
+                ...(txOptions?.maxPriorityFeePerGas && { maxPriorityFeePerGas: txOptions.maxPriorityFeePerGas })
+            };
+
+            console.time('RPC Calls Time');
+            const [gasLimit, baseGasPrice] = await Promise.all([
+                !tx.gasLimit ? signer.estimateGas({
+                    ...tx,
+                    gasPrice: ethers.utils.parseUnits('1', 'gwei')
+                }) : Promise.resolve(tx.gasLimit),
+                (!tx.gasPrice && !tx.maxFeePerGas) ? signer.provider!.getGasPrice() : Promise.resolve(undefined)
+            ]);
+            console.timeEnd('RPC Calls Time');
+
+            if (!tx.gasLimit) {
+                tx.gasLimit = gasLimit;
+            }
+
+            if (!tx.gasPrice && !tx.maxFeePerGas && baseGasPrice) {
+                if (txOptions?.priorityFee) {
+                    const priorityFeeWei = ethers.utils.parseUnits(
+                        txOptions.priorityFee.toString(),
+                        'gwei'
+                    );
+                    tx.gasPrice = baseGasPrice.add(priorityFeeWei);
+                } else {
+                    tx.gasPrice = baseGasPrice;
+                }
+            }
+
+            console.time('Transaction Send Time');
+            const transaction = await signer.sendTransaction(tx);
+            console.timeEnd('Transaction Send Time');
+
+            console.time('Transaction Wait Time');
+            const receipt = await transaction.wait(1);
```

Comment:
is 1 confirmation ok or should we add it as an option?

Reply:
Default is 1, I will remove this